os/board/common/partitions.c

```diff
@@ -317,9 +326,29 @@ void automount_fs_partition(partition_info_t *partinfo)
 		ret = mount(fs_devname, "/mnt", "smartfs", 0, NULL);
 		if (ret != OK) {
 			lldbg("ERROR: mounting '%s' failed, errno %d\n", fs_devname, get_errno());
+
+#if defined(CONFIG_LIB_BOARDCTL) && defined(CONFIG_BOARDCTL_RESET)			
+			/* Mount was failed, so we'll try low level format forcely */
+			lldbg("Low level format will be done, then board will reset\n");
+
+#ifdef CONFIG_SMARTFS_MULTI_ROOT_DIRS
+			ret = mksmartfs(fs_devname, 1, true);
+#else
+			ret = mksmartfs(fs_devname, true);
+#endif
+			if (ret != OK) {
+				lldbg("ERROR: mksmartfs on %s failed again\n", fs_devname);
+				return;
+			} 
+			/* After llformat, Reset to free all resources & initialialize smartfs dev */
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+			prctl(PR_REBOOT_REASON_WRITE, REBOOT_SYSTEM_USERFS_FORMAT);
```

Comment:
This is os code, so you can use `up_reboot_reason_write(REBOOT_SYSTEM_USERFS_FORMAT);` directly.